File: src/NullSafety.java
public class NullSafety {

    public static void main(String[] args) {
        Company company = new Company();
        Integer length = 0;
        if (company != null) {
            if (company.department != null) {
                if (company.department.team != null) {
                    length = company.department.team.name.length();
                }
            }
        }
    }

    private static class Company {
        public Department department;
    }

    private class Department {
        public Team team;
    }

    private class Team {
        public String name;
    }
}

File: kotlin-examples/src/main/kotlin/NullSafety.kt
fun main(args: Array<String>) {
    var nullSafename: String = "Kotlin"
    nullSafename.length // guaranteed to not cause an NPE
    // nullSafename = null // compilation error

    var nullableName: String? = "Kotlin"
    //nullableName.length // compilation error
    nullableName = null

    val company = Company()
    val length: Int? = company?.department?.team?.name?.length

    val name: String? = null
    val nameLength: Int = name?.length ?: -1
}


private class Company {
    var department: Department? = null
}

private class Department {
    var team: Team? = null
}

private class Team {
    var name: String? = null
}